checker/checker.go

```diff
@@ -8,7 +8,7 @@ import (
 	"reflect"
 )
 
-func Check(tree *parser.Tree, types TypesTable) (t reflect.Type, err error) {
+func Check(tree *parser.Tree, types TypesTable, operators map[string][]string) (t reflect.Type, err error) {
```

Comment:
Let's split type `map[string][]string` to smaller types with comments of what they are containing.